pypytools/pypylog/testing/test_parse.py

```diff
@@ -33,7 +41,7 @@ def test_mismatch(self):
         [456] foo}
         [0ab] bar}
         """
-        pytest.raises(parse.ParseError, "self.parse(log)")
+        pytest.raises(parse.ParseError(self.parse(log)))
```

Comment:
@a16bitsysop this change is incorrect — it should be a context manager. Otherwise, the invocation happens before `pytest.raises()` is called and it won't work.

Reply:
This is what's causing the error @ https://github.com/antocuni/pypytools/pull/2#issuecomment-1005141592.

Reply:
```suggestion
        with pytest.raises(
            parse.ParseError,
            match=r'^End section does not match start: expected bar, got foo$',
        ):
            self.parse(log)
```